python/ray/rllib/agents/agent.py

```diff
@@ -66,8 +66,13 @@
     # === Environment ===
     # Discount factor of the MDP
     "gamma": 0.99,
-    # Number of steps after which the episode is forced to terminate
+    # Number of steps after which the episode is forced to terminate. Defaults
+    # to `env.spec.max_episode_steps` (if present) for Gym envs.
     "horizon": None,
+    # Calculate rewards but don't reset the environment when the horizon is
+    # hit. This allows value estimation and RNN state to span across logical
+    # episodes denoted by horizon. This only has an effect if horizon != inf.
+    "soft_horizon": False,
```

Comment:
Better name?

Reply:
Maybe change `horizon -> max_episode_length` and `soft_horizon -> horizon`?

Reply:
Yeah that's a bit clearer. It might not be worth renaming horizon thou, we'd have to deprecate it.